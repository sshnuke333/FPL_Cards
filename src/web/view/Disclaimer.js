import React from 'react';
import {
    DisclaimerDiv,
    ExternalLink,
    Legal,
    StyledP,
} from './Disclaimer.styles';

export const Disclaimer = () => {
    return (
        <DisclaimerDiv>
            <Legal>Legal</Legal>
            <StyledP color="#dad400">
                This App is not a licensed Premier League product.
            </StyledP>
            <StyledP>
                This app uses data which is protected by copyright, database
                rights and other intellectual property and related rights owned
                by{' '}
                <ExternalLink
                    href="https://www.premierleague.com/"
                    target="_blank"
                >
                    Premier League
                </ExternalLink>{' '}
                and it's suppliers. Usage of these copyrighted works were not
                specifically authorized by{' '}
                <ExternalLink
                    href="https://www.premierleague.com/"
                    target="_blank"
                >
                    Premier League
                </ExternalLink>{' '}
                or it's suppliers but is being used fairly in good faith for
                non-commercial research study.
            </StyledP>
            <StyledP>
                This app uses{' '}
                <ExternalLink
                    href="https://www.fontfabric.com/fonts/uni-sans/"
                    target="_blank"
                >
                    Uni Sans Heavy
                </ExternalLink>{' '}
                font, a freeware provided by Svetoslav Simov / fontfabric.{' '}
            </StyledP>
        </DisclaimerDiv>
    );
};
