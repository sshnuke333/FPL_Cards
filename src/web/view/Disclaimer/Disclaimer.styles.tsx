import styled from 'styled-components';

export const DisclaimerDiv = styled.div`
    display: grid;
    place-items: start;
    margin: 2rem;
    padding: 2rem;
    color: white;
    font-size: 2rem;
    line-height: 1.5;
    @media only screen and (max-width: 768px) {
        font-size: 1.5rem;
    } ;
`;

export const ExternalLink = styled.a`
    color: #dad400;
`;

export const Legal = styled.h2`
    color: #dad400;
    font-size: 3rem;
`;

export const StyledP = styled.p`
    color: ${({ color }) => color ?? 'white'};
    padding: 0.5rem 0;
`;
