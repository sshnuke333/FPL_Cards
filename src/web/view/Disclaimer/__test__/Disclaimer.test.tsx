import { render } from '@testing-library/react';
import Disclaimer from '../Disclaimer';

describe('Disclaimer', () => {
    it('renders correctly', () => {
        const { asFragment } = render(<Disclaimer />);
        expect(asFragment()).toMatchSnapshot();
    });
});
