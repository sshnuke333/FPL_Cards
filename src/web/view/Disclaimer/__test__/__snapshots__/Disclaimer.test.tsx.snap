// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Disclaimer renders correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bcXHqe hsmusz"
  >
    <h2
      class="sc-dkrFOg hksEhS"
    >
      Legal
    </h2>
    <p
      class="sc-hLBbgP rwtwX"
      color="#dad400"
    >
      This App is not a licensed Premier League product.
    </p>
    <p
      class="sc-hLBbgP ehJNxS"
    >
      This app uses data which is protected by copyright, database rights and other intellectual property and related rights owned by 
      <a
        class="sc-gswNZR gpWUOx"
        href="https://www.premierleague.com/"
        target="_blank"
      >
        Premier League
      </a>
       and it's suppliers. Usage of these copyrighted works were not specifically authorized by 
      <a
        class="sc-gswNZR gpWUOx"
        href="https://www.premierleague.com/"
        target="_blank"
      >
        Premier League
      </a>
       or it's suppliers but is being used fairly in good faith for non-commercial research study.
    </p>
    <p
      class="sc-hLBbgP ehJNxS"
    >
      This app uses 
      <a
        class="sc-gswNZR gpWUOx"
        href="https://www.fontfabric.com/fonts/uni-sans/"
        target="_blank"
      >
        Uni Sans Heavy
      </a>
       font, a freeware provided by Svetoslav Simov / fontfabric. 
    </p>
  </div>
</DocumentFragment>
`;
